department-of-veterans-affairs/caseflow
# Shalliburton/caseflow 1078 distribute hearing days evenly

This is a continuation of the work from [this PR](https://github.com/department-of-veterans-affairs/caseflow/pull/15965) that was closed because we went in a different direction with the algorithm

Resolves #{github issue number}

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] When a user uploads the RO/CO hearings spreadsheet for a given date range, then the number of hearing days created for any date is close to the average (we can make that more specific)
- [ ] (total # of hearing days without rooms across all ROs) / (available days in the range) rounded up
- [ ] (total # of hearing days without rooms across all ROs) / (total # of weekdays in range - holidays - # of Board non-availability dates) rounded up

### Testing Plan
1. Shadow user `BVATWARNER` and navigate to the build schedule page: http://localhost:3000/hearings/schedule/build
2. Click the `Upload Files` button
3. Set the start date to 04/01/2021 and the end date to 06/30/2021
4. Upload this file: [RO Assignment Test Q3 FY21 .xlsx](https://github.com/department-of-veterans-affairs/caseflow/files/6461629/RO.Assignment.Test.Q3.FY21.xlsx)
5. Ensure that the algorithm runs successfuly

**Create the Distributino Pivot Table**
1. Once the algorithm runs successfully and you are presented with the results in a table, copy everything including the headers and paste into a new Excel workbook
2. In Excel, with the whole data range selected (headers included) go to the "Insert" tab and click on the "Pivot Table". Add the table to a new worksheet to start building the pivot table
3. Drag the Regional Office to the columns, the Date to the rows and the values (Count) and then drag the rooms to the filter
4. Apply the filter removing all rooms except the blank ones
5. View the results, make sure to expand the months to see each days distribution

# Notes

- Most Recent Distribution Results: [NoRoomHearingDayAssignments .xlsx](https://github.com/department-of-veterans-affairs/caseflow/files/6468923/NoRoomHearingDayAssignments.xlsx)
